exporter/keys.go

```diff
@@ -115,6 +115,37 @@ func getKeysFromPatterns(c redis.Conn, keys []dbKeyPair) (expandedKeys []dbKeyPa
 	return expandedKeys, err
 }
 
+func (e *Exporter) recordCheckKeyMetric(k dbKeyPair, ch chan<- prometheus.Metric, c redis.Conn, defaultValue ...float64) {
+	if _, err := doRedisCmd(c, "SELECT", k.db); err != nil {
+		log.Debugf("Couldn't select database %#v when getting key info.", k.db)
+		return
+	}
+
+	dbLabel := "db" + k.db
+	info, err := getKeyInfo(c, k.key)
+	if err != nil {
+		switch err {
+		case errKeyTypeNotFound:
+			if len(defaultValue) > 0 {
```

Comment:
Ah -- checking len(defaultValue) might be less an interesting pattern and more unfamiliarity with Go: it was my approach to setting an optional value for `defaultValue` so that we don't need to duplicate the code in `recordCheckKeyMetric` in order to handle both setting a default (check-single-keys) and omitting missing keys (check-keys). I'm handling those differently because (as discussed in #472) we:
- don't want to add the overhead of keeping track of previously seen keys
- won't know about keys from check-keys until the nth scrape

I'd be keen to hear your thoughts on more idiomatic ways to approach this in Go (if any).

You're right that we don't record zero for keys in `scannedKeys`, only for the keys in `singleKeys`. Probably not too much harm in recording 0 anyways but might be confusing to users if _sometimes_ it records 0 for keys found by check-keys and sometimes not (however rare that might be). This way, we can say:
- `check-single-keys`: defaults to 0 if not found
- `check-keys`: will be missing if not found

That said, I'm happy to go either way, just let me know which you prefer. If we're always writing a default value of 0 I'll probably scrap the `recordCheckKeyMetrics` function and go back to the for loop that it is replacing.

Reply:
Sorry for the delay.
I think in that case you can maybe just simplify this and always use the default value of zero if the key was not found.

Reply:
Sounds good! Cleaned up the commit history and addressed in 38d2106 -- we now default to 0 all the time.